index.html

```diff
@@ -207,99 +207,104 @@ <h3>
-        <a data-xref-for="">browsing context</a> that is exists in the
-        {{DedicatedWorkerGlobalScope}}'s <a>owner set</a>.
-      </div>
+      <section data-dfn-for="WakeLockPermissionDescriptor">
+        <h2>
+          The <dfn>WakeLockPermissionDescriptor</dfn> dictionary
+        </h2>
+        <p>
+          The `"wake-lock"` <a>powerful feature</a> is defined as follows:
+        </p>
+        <p>
+          The <a>permission descriptor type</a> is represented by the
+          <a>WakeLockPermissionDescriptor</a> <a>dictionary</a>, which
+          extends the {{PermissionDescriptor}} with a <dfn>type</dfn> member,
+          allowing for more fine-grained permissions.
+        </p>
+        <pre class="idl">
+          dictionary WakeLockPermissionDescriptor : PermissionDescriptor {
+            WakeLockType type;
+          };
+        </pre>
+        <p>
+          The {{PermissionDescriptor.name}} is `"wake-lock"`.
+        </p>
+      </section>
+      <section>
+        <h2>
+          Permission algorithms
+        </h2>
+        <p>
+          To <dfn>block a permission</dfn>, run these steps:
+        </p>
+        <ol class="algorithm">
+          <li>Let |descriptor:WakeLockPermissionDescriptor| be an instance of the
+            {{WakeLockPermissionDescriptor}} to be blocked.
+          </li>
+          <li>Set |descriptor|'s <a>permission state</a> to {{"denied"}}.
+          </li>
+          <li>Let |record| be the <a>platform wake lock</a>'s <a>state record</a>
+          associated with |descriptor|'s <a data-lt=
+          "WakeLockPermissionDescriptor.type">type</a> member.
+          </li>
+          <li>[=list/For each=] |lock:WakeLock| in |record|.{{[[InstanceList]]}}:
+            <ol>
+              <li>Run <a>release a wake lock</a> on |lock|.
+              </li>
+            </ol>
+          </li>
+        </ol>
+        <p>
+          To <dfn>obtain permission</dfn> for <a>wake lock type</a> |type|, run
+          these steps <a>in parallel</a>. This async algorithm returns either
+          {{"granted"}} or {{"denied"}}.
+        </p>
+        <ol class="algorithm">
+          <li>Let |permissionDesc:WakeLockPermissionDescriptor| be a newly
+          created {{WakeLockPermissionDescriptor}}.
+          </li>
+          <li data-cite="permissions">Set |permissionDesc|'s {{name}} member to
+          "`wake-lock`".
+          </li>
+          <li>Set |permissionDesc|'s {{type}} to |type|.
+          </li>
+          <li>Let |resultPromise:Promise| be the result of running <a>query a
+          permission</a> with |permissionDesc|.
+          </li>
+          <li>Await |resultPromise| to settle.
+          </li>
+          <li>If |resultPromise| rejects, return {{"denied"}}.
+          </li>
+          <li>Otherwise, let |status:PermissionStatus| be the result of
+          |resultPromise|.
+          </li>
+          <li>Let |state:PermissionState| be the value of |status|.<code><a data-cite=
```

Comment:
you should be able to do  `<li data-cite="permissions">`... less ugly.

Reply:
Had to add  data-xref-for="PermissionStatus" as well

Reply:
I'll take a look once this is all up.